zulipterminal/ui_tools/boxes.py

```diff
@@ -365,17 +365,27 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
         if not reactions:
             return ''
         try:
-            reaction_stats = Counter()  # type: typing.Counter[str]
+            reaction_stats = defaultdict(set)
             for reaction in reactions:
-                reaction_stats[reaction['emoji_name']] += 1
+                user_id = int(reaction['user'].get('id', -1))
+                if user_id == -1:
+                    user_id = int(reaction['user']['user_id'])
```

Comment:
Why not something like -
```
  user_id = reaction['user']['id'] 
           if 'id' in reaction['user'] 
           else reaction['user']['user_id']
```

Reply:
Regarding this version of the code, I follow advice I was given previously and agree with, which is that if these conditional expressions go beyond one line, then it's probably unclear - or at least equivalent to just using something like I did, or equivalently a version of what you did:
```python3
if 'id' in reaction['user']:
    user_id = reaction['user']['id']
else:
    user_id = reaction['user']['user_id']
```
I would claim that is clearer than the conditional (if an extra line), and we could indeed use it...

... but that said, the real solution here is to do the normalization of the `user` dict in the first place, which will make this kind of check go away completely!

Reply:
Right, fixing the mismatch in the events should render this unnecessary. :+1: